infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/RemoveProjectOnWorkspaceRemove.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.project;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.eclipse.che.api.core.notification.EventService;
+import org.eclipse.che.api.core.notification.EventSubscriber;
+import org.eclipse.che.api.workspace.server.event.WorkspaceRemovedEvent;
+import org.eclipse.che.workspace.infrastructure.openshift.OpenShiftClientFactory;
+
+/**
+ * Listener for removing OpenShift project on {@code WorkspaceRemovedEvent}.
+ *
+ * @author Sergii Leshchenko
+ */
+@Singleton
+public class RemoveProjectOnWorkspaceRemove implements EventSubscriber<WorkspaceRemovedEvent> {
+
+  private final OpenShiftClientFactory clientFactory;
+
+  @Inject
+  public RemoveProjectOnWorkspaceRemove(OpenShiftClientFactory clientFactory) {
+    this.clientFactory = clientFactory;
+  }
+
+  @Inject
+  public void subscribe(EventService eventService) {
```

Comment:
We usually do this with post construct annotation. Can you elaborate on how this solution differs?

Reply:
@PostConstuct annotation is functionality that is implemented by ourselves https://github.com/eclipse/che/blob/6e9825c699d34eef6b8bf36fc0194ff871cbee84/core/commons/che-core-commons-inject/src/main/java/org/eclipse/che/inject/CheBootstrap.java#L135

Using of @Inject annotation on methods is functionality that is provided by Guice.

As we don't need EventService outside of this method I prefer to inject EventService instance into the method.